### PR TITLE
Fix minigrid policy type and add new envs

### PR DESCRIPTION
**Pull request by qgallouedec**
## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by araffin** [comment]
> @araffin why does minigrid need `gym-packages` arg? If there is no specific reason, I would use `import_envs.py`, like for other libraries.

mainly for legacy reasons and to demonstrate how to use `--gym-packages`.

**Comment by qgallouedec** [comment]
Would you agree to remove it?

**Comment by araffin** [comment]
> Would you agree to remove it?

Remove the argument completely, no. It is actually useful from time to time (even though the feature is now mostly covered by python config now), and we should probably rename it to `packages`.

Change the way minigrid is used?
yes, but then we should probably add a test for the `gym-packages` to avoid regressions.

**Comment by qgallouedec** [comment]
I mean, changing the way that minigrid is used, sorry
I agree with your remark about testing ˋgym-packages`

**Comment by qgallouedec** [comment]
https://wandb.ai/qgallouedec/sb3-minigrid

**Review by araffin — DISMISSED** [review]
LGTM, thanks =)

**Comment by qgallouedec** [comment]
I just wait to have one run per env (and maybe adjust n_timesteps) before merging

**Comment by qgallouedec** [comment]
I think that we're good now

**Comment by qgallouedec** [comment]
Note that I did not run the benchmark script. It fails during the evaluation Pendulum/PPO, but I did not investigate further.

**Comment by qgallouedec** [comment]
Note that I did not run the benchmark script. It fails during the evaluation Pendulum/PPO, but I did not investigate further.

File: README.md [hunk 1]
```diff
@@ -235,31 +235,18 @@ We used the v1 environments.
...
+| A2C     |                    |                    |                    |                    |                    |                    |                    |                    |                    |                    |                    |                     |
+| PPO     | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark:  |
+| DDPG    |                    |                    |                    |                    |                    |                    |                    |                    |                    |                    |                    |                     |
+| SAC     |                    |                    |                    |                    |                    |                    |                    |                    |                    |                    |                    |                     |
```
Comment (araffin) [inline README.md:246]: does it make sense for DDPG/SAC/TD3? I thought those env have discrete actions only?

Comment (qgallouedec) [reply README.md:246]: Indeed, I replaced SAC and DDPG by DQN and QR-DQN

File: hyperparams/ppo.yml [hunk 2]
```diff
@@ -284,15 +284,14 @@ InvertedPendulumSwingupBulletEnv-v0:
...
   clip_range: 0.2
 
+
 # Following https://github.com/lcswillems/rl-starter-files
-# requires --gym-packages gym_minigrid
-MiniGrid-DoorKey-5x5-v0:
-  # Dict Observations are now supported
-  # env_wrapper: gym_minigrid.wrappers.FlatObsWrapper
+MiniGrid-Empty-Random-5x5-v0: &minigrid-defaults
+  env_wrapper: gym_minigrid.wrappers.FlatObsWrapper # See GH/1320#issuecomment-1421108191
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply hyperparams/ppo.yml:290]: In minigrid, the observation is a dict with a single key ("image"). The value is an array of shape (7, 7, 3) and dtype uint8. So it is properly detected as an image. However, it is a "false image" (at least, I don't think we want to consider it as such, see 1320).

There are two possibilities (both work):
- The one I use in this branch: we use FlatObsWrapper to convert the observation to a vector, we use MlpPolicy as the policy type.
- The one in the master branch: we don't use FlatObsWrapper, so the observation is a dict. The type of extractor will therefore be chosen automatically. As the observation is normalized (`normalize: true`), its dtype changes (uint8 to float32), and it is not considered as an image anymore. Thus, the extractor used is `MlpExtractor`.

The problem with the second option is that if one day I want to evaluate the impact of the normalization, and I set `normalize: false`, I expect that only the normalization will change. But what will happen is that the observation will be detected as an image and the extractor used will be `NatureCNN`.

This is why I think the first option is better.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -47,6 +47,8 @@ jobs:
         pip install opencv-python-headless
         # install parking-env to test HER (pinned so it works with gym 0.21)
         pip install highway-env==1.5.0
+        # Install a test env to test gym-packages arg
```
Comment (araffin) [inline .github/workflows/ci.yml:50]: maybe add those instructions somewhere?

Comment (qgallouedec) [reply .github/workflows/ci.yml:50]: about how to install the test package?

Comment (araffin) [reply .github/workflows/ci.yml:50]: yes because without that you cannot run the tests locally, no?

Comment (qgallouedec) [reply .github/workflows/ci.yml:50]: True! Where would you add the instructions?

Comment (araffin) [reply .github/workflows/ci.yml:50]: Either readme or in the doc (there should be a section on testing the package, if not, you can create one)

**Comment by araffin** [comment]
fyi, I removed the need for installing test env package and custom mypy arg in https://github.com/DLR-RM/rl-baselines3-zoo/pull/357/commits/666abb320ef311c99726b8448cf03b08c7c2dbc1

**Comment by qgallouedec** [comment]
It's much better like this

**Pull request merged**